cinch/library/jenkins_cli_user.py

```diff
@@ -40,56 +43,49 @@ def main():
     module = AnsibleModule(
         argument_spec={
             'jenkins_home': {'required': True},
+            'jenkins_user': {'required': True},
             'key_file': {'required': True},
             'state': {'choices': ['present'], 'default': 'present'}
         },
         supports_check_mode=False
     )
     params = type('Params', (object,), module.params)
-    jenkins_config = os.path.join(params.jenkins_home, "config.xml")
-    if not os.path.exists(jenkins_config):
-        module.exit_json(changed=False, msg='No config.xml found')
     user_config_path = os.path.join(params.jenkins_home, "users")
-    tree = ET.parse(jenkins_config)
-    root = tree.getroot()
-    roles = root.getiterator("role")
     changed = False
-    if roles is not None:
-        for role in roles:
-            name = role.attrib.get("name")
-            if name == "admin":
-                with open(params.key_file) as key:
-                    pub_key = key.read()
-                for sid in role.getiterator("sid"):
-                    user_cfg_file = os.path.join(user_config_path,
-                                                 sid.text, "config.xml")
-                    usertree = ET.parse(user_cfg_file)
-                    userroot = usertree.getroot()
-                    keyroot = userroot.find("properties")
-                    keys = keyroot.getiterator("authorizedKeys")
-                    if keys:
-                        for key in keys:
-                            if pub_key not in str(key.text):
-                                changed = True
-                                if key.text is None:
-                                    key.text = pub_key
-                                else:
-                                    key.text = str(key.text) + pub_key
+    with open(params.key_file) as key:
```

Comment:
Let's add a comment here that explains the intent of this block, and especially detail under what conditions we except things to be in a CHANGED state.